metals/src/main/scala/scala/meta/internal/pantsbuild/commands/ZipkinUrls.scala

```diff
@@ -0,0 +1,52 @@
+package scala.meta.internal.pantsbuild.commands
+
+import java.nio.charset.StandardCharsets
+import java.nio.file.{Files, Paths, StandardOpenOption}
+
+object ZipkinUrls {
+
+  def url: Option[String] =
+    Option(System.getProperty("metals.zipkin.server.url"))
+
+  def updateZipkinServerUrl(): Boolean = {
+    ZipkinUrls.url match {
+      case Some(newUrl) =>
+        val homedir = Paths.get(System.getProperty("user.home"))
+        val bloopJvmOpts = homedir.resolve(".bloop").resolve(".jvmopts")
+        val valueAssignment = s"-Dzipkin.server.url=$newUrl"
+        val linesToWrite = if (Files.exists(bloopJvmOpts)) {
+          val text =
+            new String(Files.readAllBytes(bloopJvmOpts), StandardCharsets.UTF_8)
+          val lines = text.linesIterator.toArray
+          val idx = lines.indexWhere(_.startsWith("-Dzipkin.server.url="))
+          if (idx >= 0) {
+            val oldUrl = lines(idx).split('=')(1)
+            // Case 1: Update an existing URL
+            if (oldUrl != newUrl) {
+              lines(idx) = valueAssignment
+              lines
+            } else {
+              // Case 2: Old and new URL are the same; break out
+              return false
+            }
+          } else {
+            // Case 3: .jvmopts exists but no old URL
+            lines :+ valueAssignment
+          }
+        } else {
+          // Case 4: .jvmopts does not exist
+          Array(valueAssignment)
+        }
+        scribe.info(s"Updating Zipkin server URL to $newUrl")
```

Comment:
We generally try to format logs like this
```suggestion
        scribe.info(s"zipkin: new server URL '$newUrl'")
```